packages/token-list/tsconfig.json

```diff
@@ -0,0 +1,10 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"outDir": "build-types",
+		"strictNullChecks": false,
+		"noImplicitAny": false
```

Comment:
Undesireable, but there are lots of errors without it. This section is particularly problematic:

https://github.com/WordPress/gutenberg/blob/20186c99697f769ed0d2153335a491aeb91e477d/packages/token-list/src/index.js#L20-L22

It could likely be rewritten to enable more strict checks if we'd like.

Reply:
Is it any better if we simply "unwrap" it, [Prepack](https://prepack.io/)-style?

```js
this.entries = ( ...args ) => this._valueAsArray.entries( ...args );
this.forEach = ( ...args ) => this._valueAsArray.forEach( ...args );
this.keys = ( ...args ) => this._valueAsArray.keys( ...args );
this.values = ( ...args ) => this._valueAsArray.values( ...args );
```

Reply:
I went ahead and got this strictly typed with some relatively minor restructuring in 106e3027d24c495cbeb695f9fab94c9a06944399

Reply:
Ohhhh, I remember this issue now. I ran into it in the past as well. I also recall it feeling like technically a bug in TypeScript, though admittedly our implementation is a pretty extreme edge case. Specifically, TypeScript doesn't detect the initialization of `_currentValue` and `_valueAsArray` in the constructor; they definitely are initialized there, but it's obscured by the fact that the initialization occurs as a side-effect of the setter associated with `this.value`.